### PR TITLE
forwarder: customize the listener port

### PR DESCRIPTION
**Pull request by Amulyam24**
By default the port is 15150, this adds the support to customize it to a specific port. Currently changes added for AWS, Azure and IBM Cloud PowerVS providers.

Fixes: #1457

### CODE REVIEW

**Review by yoheiueda — COMMENTED** [review]
I think `FORWARDER_ADDRESS` can also be configuration.

How about providing `FORWARDER_LISTEN` or `FORWARDER_LISTEN_ADDR` instead of `FORWARDER_PORT` only?

The default value of `FORWARDER_LISTEN` will be `0.0.0.0:15150`.

**Comment by Amulyam24** [comment]
> The default value of `FORWARDER_LISTEN` will be `0.0.0.0:15150`.

Hi @yoheiueda, are you suggesting we pass `FORWARDER_LISTEN`(addr+port) by default to the agent-protocol-forwarder?

> I think `FORWARDER_ADDRESS` can also be configuration.

Okay

**Comment by yoheiueda** [comment]
@Amulyam24 

My point was that it is preferable to make the `0.0.0.0` part configurable in addition to the  `15150` part. 

>  are you suggesting we pass FORWARDER_LISTEN(addr+port) by default to the agent-protocol-forwarder?

So, when a user want to use the default `0.0.0.0:15150`, I don't think we need to explicitly specify the value in the command line argument of `agent-protocol-forwarder`.

**Review by yoheiueda — APPROVED** [review]
LGTM.  Thank you very much!

**Comment by bpradipt** [comment]
Is using a customisable listen address (not port) applicable ? Since CAA uses the node IP (public or private) and the port to connect to the APF. 
@yoheiueda @Amulyam24  is there a scenario that I'm missing?

**Comment by Amulyam24** [comment]
> applicable ? Since CAA uses the node IP (public or private) and the port to connect to the APF.

That's right @bpradipt. I missed this point. I will let @yoheiueda answer it if he has any specific scenario case for it

**Comment by yoheiueda** [comment]
@bpradipt @Amulyam24 Oh, that's right. I don't come up with a specific scenario. I think we can go ahead without `FORWARDER_ADDRESS`.

When a pod VM image supports multiple network interfaces in the future, we will possibly provide an option to select a specific network address. But, the current deployment mechanism only supports a single network, so there are no appropriate value to be set to`FORWARDER_ADDRESS`.

**Comment by bpradipt** [comment]
@Amulyam24 can you please fix the commit check

File: podvm/Makefile.inc [hunk 1]
```diff
@@ -32,6 +32,10 @@ RUST_TARGET := $(RUST_ARCH)-unknown-linux-$(LIBC)
 CDH_RESOURCE_PROVIDER ?= kbs
 SEALED_SECRET ?= yes
 
+export FORWARDER_PORT ?=
+export FORWARDER_ADDRESS := 0.0.0.0
+export FORWARD_LISTEN = $(FORWARDER_ADDRESS):$(FORWARDER_PORT)
+
```
Comment (bpradipt) [inline podvm/Makefile.inc:38]: Do we really need these variables in the Makefile? Can't we just use this as env variable for packer to pickup ?

Comment (Amulyam24) [reply podvm/Makefile.inc]: Yes, they are not needed here. Moved to PowerVS specific Makefile and modifies packer variables accordingly, PTAL!

Comment (Amulyam24) [reply podvm/Makefile.inc]: Looking into the packer failures

**Comment by Amulyam24** [comment]
Hi @bpradipt, did you get a chance to verify this option for AWS/Azure?

**Comment by bpradipt** [comment]
@Amulyam24 sorry this PR got missed. Can you please rebase and resolve the conflicts ?

File: ibmcloud-powervs/image/Makefile [hunk 2]
```diff
@@ -5,6 +5,10 @@
 SKOPEO_SRC =
 SKOPEO_BIN = /usr/bin/skopeo
 
+FORWARDER_ADDRESS := 0.0.0.0
+export FORWARDER_PORT ?=
+export FORWARDER_LISTEN = $(FORWARDER_ADDRESS):$(FORWARDER_PORT)
+
```
[the remark this answers is not in the document]
Comment (bpradipt) [reply ibmcloud-powervs/image/Makefile:11]: FORWARDER_ADRESS is not needed - https://github.com/confidential-containers/cloud-api-adaptor/pull/1460#issuecomment-1742326305
Only the port is configurable.

[the remark this answers is not in the document]
Comment (Amulyam24) [reply ibmcloud-powervs/image/Makefile:10]: Right, the `FORWARDER_ADDRESS`, missed removing it here. Addressed, PTAL.

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @Amulyam24

**Review by stevenhorsman — APPROVED** [review]
Looks okay to me.

**Comment by Amulyam24** [comment]
Rebased the PR to resolve conflicts.

**Pull request merged**